ambry-clustermap/src/main/java/com.github.ambry.clustermap/CompositeClusterManager.java

```diff
@@ -88,6 +88,32 @@ public PartitionId getPartitionIdFromStream(InputStream stream) throws IOExcepti
   }
 
   /**
+   * Get all partition ids from both the underlying {@link StaticClusterManager} and the underlying
+   * {@link HelixClusterManager}. Compare the two and if there is a mismatch, update a metric.
+   * @return a list of partition ids from the underlying {@link StaticClusterManager}.
+   */
+  @Override
+  public List<PartitionId> getAllPartitions() {
+    List<PartitionId> staticPartitionIds = staticClusterManager.getAllPartitions();
+    if (helixClusterManager != null) {
+      Set<String> staticPartitionIdStrings = new HashSet<>();
```

Comment:
Minor: the comparison in this function is very similar to that in `getWriteablePartitionIds()`. Could we factor it out into a method like this: 
```java
static boolean partitionsEqual(List<PartitionId> partitionListOne, List<PartitionId> partitionListTwo) {
  Set<String> partitionStringsOne = new HashSet<>();
  for (PartitionId partitionId : partitionListOne) {
    partitionStringsOne.add(partitionId.toString());
  }
  Set<String> partitionStringsTwo = new HashSet<>();
  for (PartitionId partitionId : partitionListTwo) {
    partitionStringsTwo.add(partitionId.toString());
  }
  return partitionStringsOne.equals(partitionStringsTwo);
}
```